cranelift/codegen/src/egraph/elaborate.rs

```diff
@@ -248,7 +269,17 @@ impl<'a> Elaborator<'a> {
                         // is a `(cost, value)` tuple; `cost` comes first so
                         // the natural comparison works based on cost, and
                         // breaks ties based on value number.
-                        best[value] = std::cmp::min(best[x], best[y]);
+                        best[value] = if use_worst {
```

Comment:
I wonder if there' a way to encapsulte this? I realize it's a bit tricky because the order relation actually changes based on `use_worst` (reserved values come "last" when picking mins, and "first" when picking maxes). Maybe something like `pick_non_reserved_with_cmp(|x, y| x <= y)` or something like that, then `picked_non_reserved_with_cmp` takes two `(Cost, Value)` tuples and has the if-ladder?

Maybe that's just the same complexity as before, I dunno; just code-golfing a bit to see if we can make it prettier :-)

Reply:
We discussed this briefly while writing, and decided on this as it's more explicit and it's used only here.